core/contracts/financial_templates/Liquidatable.sol

```diff
@@ -232,6 +232,15 @@ contract Liquidatable is PricelessPositionManager {
         // Enqueue a request with the DVM.
         _requestOraclePrice(disputedLiquidation.liquidationTime);
 
+        // Pay a final fee
+        StoreInterface store = StoreInterface(finder.getImplementationAddress("Store"));
+        FixedPoint.Unsigned memory finalFee = store.computeFinalFee(address(collateralCurrency));
+        if (finalFee.isGreaterThan(0)) {
+            require(collateralCurrency.transferFrom(msg.sender, address(this), finalFee.rawValue));
+            collateralCurrency.safeIncreaseAllowance(address(store), finalFee.rawValue);
+            store.payOracleFeesErc20(address(collateralCurrency));
+        }
+
```

Comment:
WDYT about reusing the payFinalFees() in the FeePayer contract (introduced in #915) to do this operation? To make it work, you may have to add a parameter like `address from` so the method knows who, if anyone to pull the fee from.

Reply:
The main consideration is that I want the `msg.sender` to transfer the `finalFee` before paying fees. This way the contract definitely has enough funds to send the final fee. 

I could check if the address passed in is either `address(0)` (or `address(this)`), then no need to call the `transferFrom` line

Reply:
Agreed, you'd have to do the check so you don't end up doing a transferFrom on yourself.

Reply:
yeah i think that makes sense ill add